app/models/search_builder/synthetic_category_limit.rb

```diff
@@ -0,0 +1,37 @@
+class SearchBuilder
+  # A search builder module to restrict to a certain SyntheticCollection.
+  # Won't do _anything_ just by being in the chain -- sets class_attributes
+  # on the SearchBuilder that can be used to turn it 'on', either on the class
+  # or even on an instance.
+  #
+  #.    synthetic_category_param : set to a string, and this will be taken
+  #.      from app params to identify a collection to limit to.  If unknown,
+  #.      will do nothing. (Should limit to 0 rows instead?)
```

Comment:
The question is "what does this module do, if the code asks it for a synthetic category that does not exist."  I actually don't plan to make that _possible_ in how I ultimately use this (another part of code is going to end up aborting things before it even gets there), but was just thinking through all the possibilities. 

If you do have this module engaged, and ask for a synthetic category that doesn't exist... I guess getting 0 hits back is maybe what would be expected to happen? So maybe I should write that code anyway to be conservative, even though I don't plan for it to be possible in what I ultimately do.